likeminds-chat-reactnative-ui/components/LMChatTextInput/index.tsx

```diff
@@ -0,0 +1,217 @@
+
+  const onChangeInput = (changedText: string) => {
+    let isFirst = false;
+    const changedLen = changedText.length;
+
+    if (changedLen < inputLength) {
+      for (let i = 0; i < parts.length; i++) {
+        const cursorPosition = selection?.end ?? 0;
+        const endPosition = parts[i].position.end;
+        if (
+          cursorPosition === endPosition &&
+          parts[i].data?.original?.match(new RegExp(/@\[(.*?)\]\((.*?)\)/))
+        ) {
+          if (i === 0) {
+            isFirst = true;
+          }
+          parts = [...parts.slice(0, i), ...parts.slice(i + 1)];
+          break;
+        }
+      }
+    }
+
+    setInputLength(changedLen);
+    onType(
+      generateValueFromPartsAndChangedText(
+        parts,
+        plainText,
+        changedText,
+        isFirst
+      )
+    );
+  };
+
+  const handleTextInputRef = (ref: TextInput) => {
+    textInput.current = ref as TextInput;
+
+    if (propInputRef) {
+      if (typeof propInputRef === "function") {
+        propInputRef(ref);
+      } else {
+        (propInputRef as MutableRefObject<TextInput>).current =
+          ref as TextInput;
+      }
+    }
+  };
+
+  return (
+    <View style={StyleSheet.flatten([defaultStyles.textInput, inputTextStyle])}>
+      <TextInput
+        {...textInputProps}
+        ref={handleTextInputRef}
+        onChangeText={onChangeInput}
+        autoFocus={autoFocus}
+        onContentSizeChange={onContentSizeChange}
+        onSelectionChange={handleSelectionChange}
+        placeholderTextColor={
+          placeholderTextColor ? placeholderTextColor : "#000"
+        }
+        style={
+          rightIcon
+            ? defaultStyles.textInputWithRightIcon
+            : defaultStyles.textInputWithoutRightIcon
+        }
+        placeholder={placeholderText}
+        autoCapitalize={autoCapitalize ? autoCapitalize : "none"}
+        keyboardType={keyboardType ? keyboardType : "default"}
+        multiline={multilineField ? multilineField : false}
+        secureTextEntry={secureText ? secureText : false}
+        editable={disabled ? disabled : true}
+      >
+        <Text>
+          {parts.map(({ text, partType, data }, index) =>
+            partType ? (
+              <Text
+                key={`${index}-${data?.trigger ?? "pattern"}`}
+                style={partType.textStyle ?? defaultMentionTextStyle}
+              >
+                {text}
+              </Text>
+            ) : (
+              children
+            )
+          )}
+        </Text>
+      </TextInput>
+      {/* icon on right of text input */}
+      {rightIcon && (
+        <LMChatButton
+          {...rightIcon}
+          onTap={rightIcon.onTap}
+          text={rightIcon.text}
+          icon={{
+            type: "png",
+            assetPath: rightIcon.icon?.assetPath,
+            ...rightIcon.icon,
+          }}
+          placement={rightIcon.placement}
+          buttonStyle={defaultStyles.rightIconButton}
+        />
```

Comment:
what is the use case of `LMChatButton` here?

Reply:
will remove it, its of no use